### PR TITLE
Update go-autorest to v9.10.0

### LINKED ISSUE #18875: Authentication for azurerm module with environment usgovernment uses incorrect AAD endpoint
### Terraform Version
```
10.8.0
```

### Expected Behavior
`terraform init` should complete successfully when being initialized with environment=usgovernment

### Actual Behavior
`terraform init` fails with the following message:
```
Error configuring the backend "azurerm": Error retrieving keys for storage account "XX": azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.usgovcloudapi.net/subscriptions/XX/resourceGroups/XX/providers/Microsoft.Storage/storageAccounts/XX/listKeys?api-version=2016-12-01: StatusCode=0 -- Original Error: adal: Refresh request failed. Status Code = '400'
```

### Steps to Reproduce
Will add shortly.

### Additional Context
This is due to an update in the Azure AD auth endpoint for Azure US Government. See [announcement](https://blogs.msdn.microsoft.com/azuregov/2018/04/03/azure-government-aad-authority-endpoint-update/). go-autorest made the appropriate update in v9.7.0 however terraform is still using an older version of go-autorest, thus the solution to this issue is to update go-autorest.

Also, according the [go-autorest's changelog notes on v9.0.0](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md#v900):

>IMPORTANT: This release was initially labeled incorrectly as v8.4.0. From the time it was released, it should have been marked v9.0.0 because it contains breaking changes to the MSI packages. We apologize for any inconvenience this causes.

Since upgrading from v8.3.0 to v.8.4.0 isn't a major update and based on the note above, upgrading from v8.4.0/v9.0.0 to v9.7.0 isn't upgrading across major releases either, this should be a low risk upgrade. Also, no breaking changes are indicated in the release that occurred between these versions.

### References
Will add shortly.

### PR DESCRIPTION
**Pull request by gsacavdm**
Update go-autorest to v9.10.0 to fix #18875 since go-autorest v9.7.0 has the required fix necessary to address the Azure AD auth endpoint update for Azure US Government. See [announcement](https://blogs.msdn.microsoft.com/azuregov/2018/04/03/azure-government-aad-authority-endpoint-update/). While we could limit the update to v.9.7.0, might as well go to the highest non-major update for that version.

As indicated in the issue, even though this upgrades from 8.3 to 9.10, this isn't really a major update.

As per which references the [go-autorest's changelog notes on v9.0.0](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md#v900):

>IMPORTANT: This release was initially labeled incorrectly as v8.4.0. From the time it was released, it should have been marked v9.0.0 because it contains breaking changes to the MSI packages. We apologize for any inconvenience this causes.

Since upgrading from v8.3.0 to v.8.4.0 isn't a major update and based on the note above, upgrading from v8.4.0/v9.0.0 to v9.7.0 isn't upgrading across major releases either, this should be a low risk upgrade. Also, no breaking changes are indicated in the release that occurred between these versions.

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -63,52 +63,58 @@
...
+			"revision": "fc3b03a2d2d1f43fad3007038bd16f044f870722",
+			"revisionTime": "2018-02-05T16:51:11Z"
+		},
+		{
+			"checksumSHA1": "1Y2+bSzYrdPHQqRjR1OrBMHAvxY=",
 			"path": "github.com/Azure/go-autorest/autorest",
-			"revision": "5432abe734f8d95c78340cd56712f912906e6514",
-			"revisionTime": "2017-08-29T19:03:17Z",
-			"version": "v8.3.1",
-			"versionExact": "v8.3.1"
```
[the remark this answers is not in the document]
Comment (gsacavdm) [reply vendor/vendor.json]: Done! Apologies, I'm relatively new to go and totally new to govendor.

**Comment by gsacavdm** [comment]
Fixed @tombuildsstuff  :)

**Comment by gsacavdm** [comment]
Ping @tombuildsstuff

**Comment by tombuildsstuff** [comment]
@gsacavdm I've pushed a commit to update the vendoring to use v9.1.0 (rather than v11.x) - hope you don't mind. Going to ask someone else internally to review this since I've pushed those changes.

**Comment by gsacavdm** [comment]
Do you mean 9.10 or 9.1? I'm assuming 9.10 which is ok given that the required change is on v9.7, so anything greater than that is ok. Do you need anything from me to make this happen?

**Comment by gsacavdm** [comment]
Hey @tombuildsstuff , any updates on getting this merged?

**Comment by paultyng** [comment]
We are planning to get this in for the next TF release, just need to finish up some testing I believe.

**Comment by tombuildsstuff** [comment]
@gsacavdm as @paultyng has mentioned this'll go out as a part of the next patch release - we just need to do some edge-case testing to confirm there's nothing else broken here (which is on my list for this week) - sorry for the delay here!

**Comment by gsacavdm** [comment]
Hurray! Thanks guys!

**Pull request merged**